File: daily/203 Remove Linked List.py
Definition for singly-linked list.
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None


    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """

        # define 3 node

        dummy = ListNode(-1)
        dummy.next = head
        current = dummy

        while current.next != None:
            if current.next.val == val:
                current.next = current.next.next
            else:
                current = current.next

        return dummy.next

File: Daily Coding Practice/2019-12-25/203 Remove Linked List.cpp
#include <iostream>
#include<string>

struct ListNode {
      int val;           // define a value
      ListNode *next;    // define a pointer
      ListNode(int x) : val(x), next(NULL) {}   // constructor
 };



class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        
        ListNode *pseudo_head = new ListNode(0);
        pseudo_head->next = head;
        ListNode *cur = pseudo_head;

        while(cur){
            if (cur->next && cur->next->val == val){
                cur->next = cur->next->next;
            }
            else cur = cur->next;
        }
        return pseudo_head->next;

    }
};

